### PR TITLE
PP-13315: Use generic ChargeException

### PR DESCRIPTION
**Pull request by oswaldquek**
Clean up exceptions in https://github.com/alphagov/pay-connector/tree/master/src/main/java/uk/gov/pay/connector/charge/exception - use ChargeException where possible.

Exceptions which take a parameter (like gateway account id, field), and exceptions which are used in multiple places were spared the cut.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/GatewayAccountResource.java [hunk 1]
```diff
@@ -287,7 +288,8 @@ public Response createNewGatewayAccount(@Valid @NotNull GatewayAccountRequest ga
             String serviceId = gatewayAccountRequest.getServiceId();
             var existingGatewayAccount = gatewayAccountService.getGatewayAccountByServiceIdAndAccountType(serviceId, accountType);
             if (existingGatewayAccount.isPresent()) {
-                throw new ConflictWebApplicationException(String.format("Gateway account with service id %s and account type '%s' already exists.", serviceId, accountType));
+                throw new ChargeException(String.format("Gateway account with service id %s and account type '%s' already exists.", serviceId, accountType),
```
Comment (DomBelcher) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/resource/GatewayAccountResource.java]: Why is this a `ChargeException`?

File: src/main/java/uk/gov/pay/connector/charge/exception/Worldpay3dsFlexJwtCredentialsException.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package uk.gov.pay.connector.charge.exception;
 
+import javax.ws.rs.WebApplicationException;
```
Comment (DomBelcher) [inline src/main/java/uk/gov/pay/connector/charge/exception/Worldpay3dsFlexJwtCredentialsException.java]: Is this import necessary?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java [hunk 3]
```diff
@@ -104,11 +106,11 @@ private GatewayAccountEntity getSandboxGatewayAccount(String serviceId) {
                 .orElseThrow(() -> new GatewayAccountNotFoundException(serviceId, TEST));
 
         if (maybeSandboxTestAccount.getGatewayName().equalsIgnoreCase("stripe")) {
-            throw new ConflictWebApplicationException("Cannot request Stripe test account because a Stripe test account already exists.");
+            throw new ChargeException("Cannot request Stripe test account because a Stripe test account already exists.", GENERIC, SC_CONFLICT);
```
Comment (DomBelcher) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: I don't think this should be a `ChargeException` either as this doesn't relate to a charge

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java [hunk 4]
```diff
@@ -104,11 +106,11 @@ private GatewayAccountEntity getSandboxGatewayAccount(String serviceId) {
...
         }
 
         if (!maybeSandboxTestAccount.getGatewayName().equalsIgnoreCase("sandbox")) {
-            throw new ConflictWebApplicationException("Cannot request Stripe test account because existing test account is not a Sandbox one.");
+            throw new ChargeException("Cannot request Stripe test account because existing test account is not a Sandbox one.", GENERIC, SC_CONFLICT);
```
Comment (DomBelcher) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: Same again re. this doesn't relate to a charge

File: src/test/java/uk/gov/pay/connector/charge/service/ChargeServiceIT.java [hunk 5]
```diff
@@ -6,6 +6,7 @@
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.connector.charge.exception.ChargeException;
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/charge/service/ChargeServiceIT.java]: is this import necessary?

File: src/test/java/uk/gov/pay/connector/charge/service/ChargeServiceIdempotencyTest.java [hunk 6]
```diff
@@ -21,6 +21,7 @@
 import uk.gov.pay.connector.app.LinksConfig;
 import uk.gov.pay.connector.cardtype.dao.CardTypeDao;
 import uk.gov.pay.connector.charge.dao.ChargeDao;
+import uk.gov.pay.connector.charge.exception.ChargeException;
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/charge/service/ChargeServiceIdempotencyTest.java]: is this import necessary?

**Review by DomBelcher — CHANGES_REQUESTED** [review]
I'm in favour of eliminating single use exception classes, however I think this might be a little overzealous. I'm not familiar enough with what Agreements and Payment Instruments are to know if it's appropriate to use `ChargeException` for them, but I don't think all instances of `ConflictWebApplicationException` can be replaced with `ChargeException`

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/GatewayAccountResource.java (hunk 1 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/resource/GatewayAccountResource.java]: Whoops... I had originally deleted `ConflictWebApplicationException` but then later decided against it so this should be reverted.

**Review by DomBelcher — APPROVED** [review]
Nice work

**Pull request merged**